### DISCUSSION TITLE
Feature request/Bug Report??? When using Nav mode, folder page behaves very unintuitively

### DISCUSSION THREAD

**Discussion by allauthors** [requarks/wiki#6646] [Error / Bug Report] [answered]
Let's start by saying, I love wiki.js and it's been great implementing it so far!  Really appreciate NGPixel and all contributors.

I'm not sure if the behavior as is, qualifies as a bug/defect, so if not a defect, consider this a feature request.  I'm using latest stable version of wiki.js in a docker container, and have it configured to use folder-style nav only.

Create a folder with some files in it
Create a page in the parent folder with the exact same name.
e.g.
```
📁/ 
┗ 📁 Departments
    ├ 📁 Engineering 
    │   └ 🗏 Page-1
    └ 🗏 Engineering

```
Click the folder: the nav pane shows the page link with the same name at the top followed by the folder's actual contents.  
- The page displayed does not change from whatever page happened to be previously showing.
- The page link does NOT show in the nav pane when in the parent folder that actually contains the page.

Click the page.  The page is displayed but the nav pane is switched to the parent folder view (which doesn't show that page in it).

My best case desired behavior would be:
When clicking the folder itself, automatically display the page with the same name (or the create page view if not page exists).
Don't display the page on it's own at all in the nav anywhere; the folder itself is the link to display that page.  
I could not find any configuration option which would produce this result

A "second best" option would be:
When clicking the page, don't send the nav pane back to the parent folder, leave it in place on the folder with the same name whose contents are showing.
I could not find any configuration option which would produce this result

**Comment by NGPixel** [comment, marked as answer]
This is being addressed in v3.

**Reply by allauthors** [reply]
That's awesome to hear.  Much appreciated.  I'll look forward to that!  

One minor additional suggestion on the best case scenario would be that a direct link to the page correctly loads that page and sets the navpane to the contents of the folder of the same name. (probably obvious, but thought it was worth mentioning).  No need to respond to this, as it's just an elaboration on the already answered question/report.  Appreciate your time!
